src/emulator/storage/rules/manager.ts

```diff
@@ -1,52 +1,72 @@
 import * as chokidar from "chokidar";
-import * as fs from "fs";
 import { EmulatorLogger } from "../../emulatorLogger";
 import { Emulators } from "../../types";
-import { FirebaseError } from "../../../error";
 import { SourceFile } from "./types";
 import { StorageRulesIssues, StorageRulesRuntime, StorageRulesetInstance } from "./runtime";
+import { RulesConfig } from "..";
 
 /**
- * Loads and maintains a {@link StorageRulesetInstance} for a given source file. Listens for
- * changes to the file and updates the ruleset accordingly.
+ * Keeps track of the rules source file and maintains a generated ruleset for one or more storage
+ * resources.
+ * */
+export interface StorageRulesManager {
+  /** Sets source file for each resource using the rules previously passed in the constructor. */
+  start: () => Promise<StorageRulesIssues>;
+
+  /** Retrieves the generated ruleset for the resource. */
+  getRuleset: (resource: string) => StorageRulesetInstance | undefined;
+
+  /**
+   * Updates the source file and, correspondingly, the file watcher and ruleset for the resource.
```

Comment:
Can you include in the comment what would cause `StorageRulesIssues` to be returned